### PR TITLE
Add Fashion Gen pre-processing.

### PR DESCRIPTION
**Pull request by boru-roylu**
1. create_fashion_gen_metadata.ipynb

- Generates metadata for inference data.
- We use the same inference data as Fashion-BERT and kaleido-BERT.

2. create_fashion_gen_retrieval_test_data.py

- Creates inference data of Fashion Gen.

3. create_fashion_gen_split.py

- Creates training and validation sets for Fashion Gen.

### CODE REVIEW

File: preprocessing/create_fashion_gen_metadata.ipynb [hunk 1]
```diff
@@ -0,0 +1,1288 @@
...
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "height": 419
+        },
+        "id": "YMCqOYW9__M9",
+        "executionInfo": {
```
Comment (santiaontanon-google) [inline preprocessing/create_fashion_gen_metadata.ipynb:99]: Are you sure you want this type of info to be checked in? there is even a link to a picture of you here.

Comment (santiaontanon-google) [reply preprocessing/create_fashion_gen_metadata.ipynb:99]: And same for several other blocks below.

File: preprocessing/create_fashion_gen_metadata.ipynb [hunk 2]
```diff
@@ -0,0 +1,1288 @@
+{
```
Comment (santiaontanon-google) [inline preprocessing/create_fashion_gen_metadata.ipynb:1]: Do you know if there is any way to see the code in this colab directly? it's quite hard to review in this format :)

Comment (santiaontanon-google) [reply preprocessing/create_fashion_gen_metadata.ipynb:1]: Also, make sure to add the header comment that you have in the other python files here too about the license, etc.

File: preprocessing/create_fashion_gen_retrieval_test_data.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+
+"""
+The first 2 rows of txt_info file.
+
+86605,86605_0,JACKETS & COATS,7,DENIM JACKETS,39,Denim-like jogg jacket in blue. Fading and whiskering throughout. Spread collar. Copper tone button closures at front. Flap pockets at chest with metallic logo plaque. Seam pockets at sides. Cinch tabs at back waistband. Single button sleeve cuffs. Tone on tone stitching.
+86605,86605_1,JACKETS & COATS,7,DENIM JACKETS,39,Denim-like jogg jacket in blue. Fading and whiskering throughout. Spread collar. Copper tone button closures at front. Flap pockets at chest with metallic logo plaque. Seam pockets at sides. Cinch tabs at back waistband. Single button sleeve cuffs. Tone on tone stitching.
+"""
+
+txt_info_filename = f'gs://mmt/raw_data/fashion_gen/full_valid_info.txt'
```
Comment (santiaontanon-google) [inline preprocessing/create_fashion_gen_retrieval_test_data.py:30]: Add a comment to replace with actual folders if someone clones this repo and they want to run this.

File: preprocessing/create_fashion_gen_retrieval_test_data.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  record_idx = 0
+  tfrecord_filename = tfrecord_basename.format(task,
+                                               record_idx,
+                                               num_records)
+  print(f'  Write to {tfrecord_filename}.')
+  writer = tf.io.TFRecordWriter(tfrecord_filename)
+
+  for i, (_, row) in enumerate(df.iterrows(), start=1):
+    # Some description in fashion bert data does not exactly match description
+    # in txt_info. The difference is very tiny, such as an extra quote symbol.<Plug>_
```
Comment (santiaontanon-google) [inline preprocessing/create_fashion_gen_retrieval_test_data.py:84]: Left over "<Plug>_" here?

File: preprocessing/create_fashion_gen_split.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import json
+
+import tensorflow as tf
+
+import utils
+
+
+txt_info = dict()
+
+path = f'gs://mmt/raw_data/fashion_gen/full_train_info.txt'
```
Comment (santiaontanon-google) [inline preprocessing/create_fashion_gen_split.py:24]: As before, add comment to update these paths if someone clones this repo

File: preprocessing/create_fashion_gen_split.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    'max_seq_length': 512,
+    'task_type': 'mmt_classification',
+    'train_data_size': len(txt_info['train']),
+    'eval_data_size': len(txt_info['valid'])
+}
+
+with tf.io.gfile.GFile(input_meta_filename, 'w') as f:
+  json.dump(meta_data, f, indent=4)
+
+for split, num_records in [('valid', 8), ('train', 128)]:
```
Comment (santiaontanon-google) [inline preprocessing/create_fashion_gen_split.py:48]: What do these "8" and "128" mean here, please add a comment to explain (try not to have magic numbers in the code)

File: preprocessing/utils.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  if int_dict is not None:
+    for k, v in int_dict.items():
+      feature[k] = _int64_feature(v)
+  return tf.train.Example(features=tf.train.Features(feature=feature))
+
+
+def get_txt_info(txt_info_filename, description_key='description'):
+  """Gets metadata of each image_id (image file).
+  
+  Returns:
```
Comment (santiaontanon-google) [inline preprocessing/utils.py:59]: nit: if you specify the "returns" field, then specify also the "args" in the comment.

**Review by santiaontanon-google — COMMENTED** [review]
Thanks Roy, some comments. I didn't review the colab yet, will do in a second pass

File: preprocessing/create_fashion_gen_metadata.ipynb (hunk 1 above)
Comment (wqfcr) [reply preprocessing/create_fashion_gen_metadata.ipynb:99]: +1 it's hard to review this file:)

File: preprocessing/utils.py (hunk 7 above)
Comment (jtainslie) [reply preprocessing/utils.py:59]: Yes `Args` should be above `Returns` in the docstring, and no need to prefix the return value with `txt_info:`.  The lines below `Returns` should be indented though.

**Review by jtainslie — APPROVED** [review]

File: preprocessing/create_fashion_gen_metadata.ipynb (hunk 1 above)
Comment (boru-roylu) [reply preprocessing/create_fashion_gen_metadata.ipynb:99]: @santiaontanon-google @wqfcr can you use the following link to review? It's the same file.
https://github.com/googleinterns/multimodal-long-transformer-2021/blob/ae21f61e1f69beeba82b7bb0ea02087bc90c02ea/preprocessing/create_fashion_gen_metadata.ipynb

**Review by santiaontanon-google — APPROVED** [review]
LGTM

**Pull request merged**